### ISSUE TITLE
CSI topo segment wrong placement to availability zone

### ISSUE THREAD

**Issue by k0wl0n** [hashicorp/nomad#12531]
### Nomad version
`Nomad v1.3.0-beta.1`

### Issue
Wrong placement availability zone for create new volume.

### Reproduction steps
i create a volume config file

```id = "couchdbdatatest"
name = "couchdbdatatest"
type = "csi"
plugin_id = "aws-ebs0"

capability {
  access_mode     = "single-node-writer"
  attachment_mode = "file-system"
}

mount_options {
  fs_type     = "ext4"
  mount_flags = ["rw"]
}

topology_request {
  required {
    topology {
      segments {
        "topology.ebs.csi.aws.com/zone" = "ap-southeast-1b"
      }
    }
  }
}
```
then run `nomad volume create couchdbdatatest.volume`

#### Expected Result
Volume created at ap-southeast-1b

#### Actual Result
Created external volume vol-01807a3df92cf8be7 with ID couchdbdatatest
vol-01807a3df92cf8be7	gp3	100 GiB	3000	125	-	2022/04/09 09:48 GMT+7	ap-southeast-1a

**Comment by tgross** [comment]
Hi @k0wl0n! Thanks for opening this bug report. Can you share your plugin job configurations as well? That might help diagnose what happened here.

**Comment by k0wl0n** [comment]
Hi @tgross thanks for reply.

this is my plugin job:
Controller

```
job "plugin-aws-ebs-controller" {
  datacenters = ["dc1"]

  constraint {
    attribute = "${attr.os.name}"
    value     = "amazon"
  }

  group "controller" {
    task "plugin" {
      driver = "docker"

      config {
        image ="amazon/aws-ebs-csi-driver:release-1.2-amazonlinux"

        args = [
          "controller",
          "--endpoint=unix://csi/csi.sock",
          "--logtostderr",
          "--v=5",
        ]
      }

      csi_plugin {
        id        = "aws-ebs0"
        type      = "controller"
        mount_dir = "/csi"
      }

      resources {
        cpu    = 256
        memory = 256
      }
    }
  }
}
```

this is my nodes 
``` 
job "plugin-aws-ebs-nodes" {
  datacenters = ["dc1"]

    constraint {
    attribute = "${attr.os.name}"
    value     = "amazon"
  }
  type = "system"

  group "nodes" {
    task "plugin" {
      driver = "docker"

      config {
        image = "amazon/aws-ebs-csi-driver:release-1.2-amazonlinux"

        args = [
          "node",
          "--endpoint=unix://csi/csi.sock",
          "--logtostderr",
          "--v=5",
        ]
        privileged = true
      }

      csi_plugin {
        id        = "aws-ebs0"
        type      = "node"
        mount_dir = "/csi"
      }

      resources {
        cpu    = 256
        memory = 256
      }
    }
  }
}
```

My nomad client version
```
nomad -v
Nomad v1.3.0-beta.1 (2eba643965c461e401fe4a3e9442fd30104a9344)
```


but im missing this config from my volume config.
and add this
``` 
capability {
  access_mode     = "single-node-writer"
  attachment_mode = "block-device"
}

parameters {
  type = "gp3"
}
```

update CSI image version  from 1.2 to 1.5
and now its working

Result output
``` 
nomad volume create couchdbdatatest.volume
Created external volume vol-0ea25005f3254c1f1 with ID couchdbdatatest3
nomad volume create couchdbdatatest.volume
Created external volume vol-040db8a3ddbbbe9a0 with ID couchdbdatatest4
nomad volume create couchdbdatatest.volume
Created external volume vol-04c91b02bf1f44da4 with ID couchdbdatatest5 

vol-04c91b02bf1f44da4	gp3	1 GiB	3000	125	-	2022/04/14 05:51 GMT+7	ap-southeast-1b	Available	
vol-040db8a3ddbbbe9a0 gp3	1 GiB	3000	125	-	2022/04/14 05:49 GMT+7	ap-southeast-1c	Available	
vol-0ea25005f3254c1f1 gp3	1 GiB	3000	125	-	2022/04/14 05:48 GMT+7	ap-southeast-1b	Available 
```

and this my final volume.hcl
``` 
id = "couchdbdatatest5"
name = "couchdbdatatest5"
type = "csi"
plugin_id = "aws-ebs0"

capability {
  access_mode     = "single-node-writer"
  attachment_mode = "file-system"
}

capacity_min = "1G"
capacity_max = "2G"

mount_options {
  fs_type     = "ext4"
  mount_flags = ["rw"]
}

parameters {
  type = "gp3"
}

topology_request {
  required {
    topology {
      segments {
        "topology.ebs.csi.aws.com/zone" = "ap-southeast-1b"
      }
    }
  }
} 
```

**Comment by tgross** [comment]
> update CSI image version from 1.2 to 1.5
> and now its working

Ok, good to hear. I know that they changed the topology key at one point but I don't see it in their [changelog](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/CHANGELOG-0.x.md) between 1.2 and 1.5, but their changelogs don't cover all updates. Ex. https://github.com/kubernetes-sigs/aws-ebs-csi-driver/commit/f6784061734899c03f248cb317f10984a8c81f8d landed in 1.5.1 without a changelog entry and that impacts topology config, so that might be what's changed. Keeping plugins on current versions is going to be important, as they're fast-moving projects.

I'm going to close out this issue as resolved by the update, but please feel free to open new issues if you run into any other issues with the 1.3.0-beta!
